### PR TITLE
Transferring llvm from svn to git

### PR DESCRIPTION
**Pull request by Sannoso**
+ fixing a small bug that caused a segfault in one of the tests.

### CODE REVIEW

File: tce/tools/scripts/install_llvm_9.sh [hunk 1]
```diff
@@ -40,33 +40,19 @@ function fetch_llvm {
...
+	git clone --branch release/9.x https://github.com/llvm/llvm-project.git $llvm_co_dir\
+	    || eexit "Git clone $REV_TOFETCH from llvm failed"
     else
-#        svn up $REV_TO_FETCH $llvm_co_dir \
-#            || eexit "SVN update $REV_TO_FETCH of LLVM failed."
-        # Wipe out possible previously applied patches.
-        svn revert -R $llvm_co_dir
+	# Discard all differences with release/9.x branch
+	cd $llvm_co_dir;
+        git checkout release/9.x ||
```
Comment (pjaaskel) [inline tce/tools/scripts/install_llvm_9.sh]: Perhaps do a git pull also to fetch any fixes to the release branch?

File: tce/tools/scripts/install_llvm_9.sh [hunk 2]
```diff
@@ -94,14 +78,13 @@ cd build
...
+cmake ../llvm/ -DLLVM_ENABLE_PROJECTS=clang \
+    -G "Unix Makefiles" \
     $LLVM_BUILD_MODE\
     -DCMAKE_INSTALL_PREFIX=$TARGET_DIR \
     -DLLVM_LINK_LLVM_DYLIB=TRUE \
     -DLLVM_ENABLE_RTTI=TRUE \
     -DLLVM_ENABLE_Z3_SOLVER=OFF \
-    .. \
     || eexit "Configuring LLVM/Clang failed."
 make -j8 CXXFLAGS="-std=c++11" REQUIRES_RTTI=1 \
```
Comment (pjaaskel) [inline tce/tools/scripts/install_llvm_9.sh]: Can you test if -std=c++11 is still needed?

File: tce/tools/patches/llvm-9-fix-load-lowering.patch [hunk 3]
```diff
@@ -0,0 +1,53 @@
+Index: llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp
```
Comment (pjaaskel) [inline tce/tools/patches/llvm-9-fix-load-lowering.patch]: What is this patch? Shouldn't it be upstreamed to LLVM?

File: tce/tools/scripts/install_llvm_9.sh (hunk 1 above)
Comment (Sannoso) [reply tce/tools/scripts/install_llvm_9.sh]: agreed. that should be done. I put it in.

File: tce/tools/scripts/install_llvm_9.sh (hunk 2 above)
Comment (hkultala) [reply tce/tools/scripts/install_llvm_9.sh]: LLVM definitely uses C++11 features. 

Whether -std=c++11 is needed or not depends on the defaults of the compiler which is used for compiling the llvm(g++ that comes with the distribution). Testing and possibly removing it from here (if works at some system without it) just causes a possible "works on my machine"-problem were it starts to fail when compiled with older/different linux distritbutions which have older version of the default g++ compiler.

Comment (pjaaskel) [reply tce/tools/scripts/install_llvm_9.sh]: OK. I know it does use the features therefore I guessed it'd add std=c++11 by default nowadays. But shouldn't harm indeed.

**Pull request merged**